herd/AArch64Sem.ml

```diff
@@ -1445,6 +1445,45 @@ module Make
           (to_perms "rw" sz) (read_reg_ord rn ii) (read_reg_data sz rt ii)
         an ii
 
+      let casp sz rmw rs1 rs2 rt1 rt2 rn ii =
+        let an = rmw_to_read rmw in
+        let (>>>) = M.data_input_next in
+        let read_rs = read_reg_data sz rs1 ii
+          >>| read_reg_data sz rs2 ii
+        and write_rs (v1,v2) = write_reg_sz_non_mixed sz rs1 v1 ii
+          >>| write_reg_sz_non_mixed sz rs2 v2 ii
+          >>= fun _ -> M.unitT () in
+        lift_memop rn Dir.W true
+          (fun ac ma _ ->
+            let is_phy = Access.is_physical ac in
+             M.altT
+              (let read_mem a = do_read_mem_ret sz an aexp ac a ii
+                >>| (add_size a sz
+                >>= fun a -> do_read_mem_ret sz an aexp ac a ii) in
+               let neqp (v1,v2) (x1,x2) =
+                 M.neqT v1 x1 >>| M.neqT v2 x2
+                 >>= fun _ -> M.unitT () in
```

Comment:
This condition is too strong: it checks v1 being different from x1 _and_ v2 being different from x2, while we probably want _or_ in place of _and_:
```suggestion
                 M.op Op.Eq v1 x1 >>| M.op Op.Eq v2 x2
                 >>= fun (b1,b2) -> M.op Op.And b1 b2
                 >>= M.eqT V.zero in

```